### PR TITLE
Profiler: Improve nested-update checks

### PR DESCRIPTION
**Pull request by bvaughn**
Previous checks were too naive when it comes to pending lower-pri work or batched updates. This commit adds two new (previously failing) tests and fixes.

### CODE REVIEW

File: packages/react/src/__tests__/ReactProfiler-test.internal.js [hunk 1]
```diff
@@ -3016,6 +3098,8 @@ describe('Profiler', () => {
         expect(Scheduler).toHaveYielded(['Component:true']);
         expect(onNestedUpdateScheduled).not.toHaveBeenCalled();
       });
+
+      // TODO Add hydration tests to ensure we don't have false positives called.
```
Comment (bvaughn) [inline packages/react/src/__tests__/ReactProfiler-test.internal.js:3102]: There might be other edge cases related to hydration or Suspense that I haven't thought about yet. This note is for me to follow up and think about them more.

**Review by acdlite — APPROVED** [review]
